sunpy/map/mapbase.py

```diff
@@ -2290,6 +2290,49 @@ def draw_quadrangle(self, bottom_left, *, width: (u.deg, u.pix) = None, height:
         quad = Quadrangle(anchor, width, height, **kwergs)
         axes.add_patch(quad)
         return quad
+    
+    def _calculate_contour_levels_by_area(self, levels):
+        """
+        Calculate contour levels based on area containment.
+
+        Parameters
+        ----------
+        levels : array_like
+            An array-like object specifying the percentages of the total area to be contained within the generated contour levels.
+            This can be a single value or an array of values representing the desired percentages, where each value should be in the range [0, 1].
+        
+        Returns
+        -------
+        thresholds : ndarray
+            An array of threshold values representing the data values below which the specified percentages of the total area are contained.
+            The length of this array is equal to the length of the input `levels`.
+        
+        Notes
+        -----
+        This function calculates contour levels based on the area contained within them relative to the maximum value in the map data.
+        It normalizes the map data, sorts the normalized values, calculates the cumulative sum, and finds thresholds corresponding to the specified percentages of the total area.
+        The thresholds are then returned as an array.
+
+        Examples
+        --------
+        >>> map_data = np.random.rand(100, 100)
+        >>> quantiles = [0.1, 0.3, 0.5, 0.7, 0.9]
+        >>> contour_levels = self._calculate_contour_levels_by_area(quantiles)
+        >>> print(contour_levels)
+        [0.3160296  0.54729944 0.70897832 0.83778233 0.95295992]
```

Comment:
I think the imports are missing? 

Also the print is not needed here. The doctest will print out the return if you do not assign it to a variable.